### ISSUE TITLE
No builder in 2.x?

### ISSUE THREAD

**Issue by kevinsmith** [relayphp/Relay.Relay#33]
I may be missing something, but without a builder in 2.x, how would you (by way of example) refactor this code in Radar? https://github.com/radarphp/Radar.Adr/blob/1.x/src/Adr.php#L137

```
    public function run(Request $request, Response $response)
    {
        $relay = $this->relayBuilder->newInstance($this->middle);
        return $relay($request, $response);
    }
```

I could write a factory of my own or set one up in my DI's definitions, but it seemed helpful in 1.x. Just curious, why was it removed?

**Comment by pmjones** [comment]
Agh -- of course. Will add it.

**Comment by cxj** [comment]
At first glance, it sort of looks like class `RelayBuilder` from the 1.x branch can be re-used almost unchanged.

**Comment by cxj** [comment]
Hah.  Silly me.  Of course, it's not that simple.  `RelayBuilder` wants a `RunnerFactory`.  Adding both of the Relay 1.x classes to the 2.x library solves that, but then somehow the type of `$queue` gets corrupted to a non-traversable.  Obviously, I need to just read all of the code.   But I was hoping to short-cut a bunch of work.  :-p

**Comment by cxj** [comment]
Looks like [Relay.Middleware](https://github.com/relayphp/Relay.Middleware) will need a 2.x version as well, at least for my use (`ResponseSender`).

**Comment by kevinsmith** [comment]
Builder has been added in #37 thanks to @jakejohns!

@cxj Yeah, looks like the entire Relay.Middleware package would need a rewrite to make them all PSR-15 compatible middleware. I've now joined the Relay project, so I'll consider it.

In the meantime, I'd recommend using an [emitter](https://github.com/narrowspark/http-emitter) to handle properly sending a response. (I suspect that's the more appropriate way of sending a response anyway.)
